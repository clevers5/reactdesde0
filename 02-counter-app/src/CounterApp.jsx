import React from 'react'
import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({value}) => {
    //Hook useState 
    const [counter, setCounter] = useState(value);

        //AÃ±adir 1 al contador
        const handleAdd = () => {
            setCounter((c) => c + 1);
        }

        //Restar 1 al contador
        const handleSubtract = () => {
            setCounter(counter - 1);
        }

        //Resetear el contador
        const handleReset = () => {
            setCounter(value);
        }

  return (
    <>
        <div>CounterApp</div>
        <h2>{counter}</h2>

        <button className='AddButton' onClick={handleSubtract}>-1</button>
        <button aria-label='btn-reset' className='ResetButton' onClick={handleReset}>Reset</button>
        <button  className='SubtractButton' onClick={handleAdd}>+1</button>
        
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 100,
}

