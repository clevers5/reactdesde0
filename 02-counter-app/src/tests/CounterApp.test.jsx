
import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../CounterApp";

const initialValue = 100;
describe('Pruebas en CounterApp', () => { 
    test('Debe hacer match con el snapshot', () => { 
        const {container} = render(<CounterApp/>);
        expect(container).toMatchSnapshot();
     })

     test('Debe de mostrar el valor inicial de 100', () => {
        
        render(<CounterApp value={100}/>);
        expect(screen.getByText(100)).toBeTruthy();
        expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain('100');
     
    });

    test('Debe de incrementar con el botón +1', () => {
    
        render(<CounterApp value={initialValue}/>);
        fireEvent.click( screen.getByText('+1'));
        expect(screen.getByText('101')).toBeTruthy();
    });

    test('Debe de decrementar con el botón -1', () => {
    
        render(<CounterApp value={initialValue}/>);
        fireEvent.click( screen.getByText('-1'));
        expect(screen.getByText('99')).toBeTruthy();
    });

    test('Debe de colocar el valor por defecto con el botón Reset', () => {
    
        render(<CounterApp value={3358}/>);
        fireEvent.click( screen.getByText('+1'));
        fireEvent.click( screen.getByText('+1'));
        // fireEvent.click( screen.getByText('Reset'));
        
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}));
        expect(screen.getByText(3358)).toBeTruthy();
    
    });
 })