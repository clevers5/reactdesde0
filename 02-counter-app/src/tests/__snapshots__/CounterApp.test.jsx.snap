// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Debe hacer match con el snapshot 1`] = `
<div>
  <div>
    CounterApp
  </div>
  <h2>
    100
  </h2>
  <button
    class="AddButton"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
    class="ResetButton"
  >
    Reset
  </button>
  <button
    class="SubtractButton"
  >
    +1
  </button>
</div>
`;
